conftest.py

```diff
@@ -393,13 +393,18 @@ def serve_wheel(request, tmp_path_factory):
         cwd=root,
     ) as prc:
         try:
-            url = f"http://localhost:{port}"
-            if existing_url := os.environ.get("PIP_EXTRA_INDEX_URL"):
-                url = f"{existing_url} {url}"
-            os.environ["PIP_EXTRA_INDEX_URL"] = url
+            localhost = f"http://localhost:{port}"
+            pip_extra_index_url = os.environ.get("PIP_EXTRA_INDEX_URL") or ""
+            os.environ["PIP_EXTRA_INDEX_URL"] = f"{pip_extra_index_url} {localhost}".strip()
             # Set the `UV_INDEX` environment variable to allow fetching the wheel from the
             # url when using `uv` as environment manager
-            os.environ["UV_INDEX"] = f"mlflow={url}"
+            extra_index = re.sub(
+                # Prepend 'pytorch=' to restrict this index to pytorch installations only.
+                r"https://download\.pytorch\.org/.*",
+                lambda m: f"pytorch={m.group(0)}",
```

Comment:
Nice! Could we merge this change into master branch first? I suffered from this error quite often :(

Reply:
I'm not sure if this really works :)

Reply:
@serena-ruan It didn't :)

Reply:
:( I tried to temporarily update PIP_EXTRA_INDEX_URL env var but it doesn't work as well 🤔